include/deal.II/base/memory_space_data.h

```diff
@@ -113,7 +116,12 @@ namespace MemorySpace
         0))
   {}
 
-
+  template <typename T, typename MemorySpace>
+  MemorySpaceData<T, MemorySpace>::MemorySpaceData(
+    Kokkos::View<T *, typename MemorySpace::kokkos_space> new_values)
+    : values_host_buffer(Kokkos::View<T *, Kokkos::HostSpace>("host buffer", 0))
+    , values(new_values)
```

Comment:
You're just copying `new_values` twice this way. Make it a `const` reference to avoid one of them.

Reply:
I don't need this constructor anymore anyway. It probably makes sense to replace `MemorySpaceData` with `Kokkos::View` at least internally in most places.